jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/test/java/org/jbpm/console/ng/cm/client/milestones/CaseMilestonesPresenterTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.console.ng.cm.client.milestones;
+
+import java.util.List;
+
+import com.google.common.collect.Lists;
+import org.jbpm.console.ng.cm.client.util.AbstractCaseInstancePresenterTest;
+import org.jbpm.console.ng.cm.client.util.CaseMilestoneStatus;
+import org.jbpm.console.ng.cm.model.CaseDefinitionSummary;
+import org.jbpm.console.ng.cm.model.CaseInstanceSummary;
+import org.jbpm.console.ng.cm.model.CaseMilestoneSummary;
+import org.jbpm.console.ng.cm.util.CaseMilestoneSearchRequest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.uberfire.mocks.CallerMock;
+
+import static java.util.Collections.*;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class CaseMilestonesPresenterTest extends AbstractCaseInstancePresenterTest {
+
+    @Mock
+    CaseMilestoneListPresenter.CaseMilestoneListView caseMilestoneListView;
+
+    @InjectMocks
+    CaseMilestoneListPresenter presenter;
+
+    @Override
+    public CaseMilestoneListPresenter getPresenter() {
+        return presenter;
+    }
+
+    List<CaseMilestoneSummary> caseMilestonesSummaryList = Lists.newArrayList(createCaseMilestone());
+    CaseInstanceSummary cis;
+    final String serverTemplateId = "serverTemplateId";
+
+    private static CaseMilestoneSummary createCaseMilestone() {
+        return CaseMilestoneSummary.builder()
+                .identifier("identifier")
+                .name("milestoneName")
+                .status(CaseMilestoneStatus.AVAILABLE.getStatus())
+                .achieved(false)
+                .build();
+    }
+
+    @Before
+    public void init() {
+        caseService = new CallerMock<>(caseManagementService);
+        when(caseManagementService.getCaseMilestonesSorted(anyString(),anyString(),any(CaseMilestoneSearchRequest.class))).thenReturn(caseMilestonesSummaryList);
+        when(caseMilestoneListView.getCaseMilestoneSearchRequest()).thenReturn(new CaseMilestoneSearchRequest());
+        presenter.setCaseService(caseService);
+
+        cis = CaseInstanceSummary.builder().build();
```

Comment:
you could use newCaseInstanceSummary instead to avoid creating an empty instance. Similar could be done for CaseDefinitionSummary if you extract a common method from here: https://github.com/droolsjbpm/jbpm-console-ng/blob/master/jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/test/java/org/jbpm/console/ng/cm/client/newcase/NewCaseInstancePresenterTest.java#L85